x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java

```diff
@@ -183,6 +187,17 @@ public boolean head() throws ClientException {
         }
     }
 
+    private Function<String, List<String>> getHeaderFields(URLConnection con) {
+        return header -> {
+            for (Map.Entry<String, List<String>> entry : con.getHeaderFields().entrySet()) {
```

Comment:
This method returns the first match - does headerFields normalizes keys? For example `Content-Type` and `ConteNT-TYPE` have their values under the same key or separate ones?